code_generator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -172,10 +177,33 @@ namespace <#= qualifiedNamespace #>
                 }
 <# } #>
                 componentData.DirtyBit = false;
+
 <# if (componentDetails.IsBlittable) { #>
-                view.UpdateComponent(entity, componentData, UpdatesPool);
+                view.SetComponentData(entity, componentData);
 <# } else { #>
-                view.UpdateComponentObject(entity, componentData, UpdatesPool);
+                view.SetComponentObject(entity, componentData);
+<# } #>
+
+<# if (fieldDetailsList.Count > 0) { #>
```

Comment:
I believe it is possible to get completely empty component update ops: updates from components that don't contain any fields. The way it's set up right now, we would simply swallow them. Do we want to add a `ComponentsUpdated` tag for these cases as well so that can be processed by any custom system? (not sure what the exact use case would be, maybe pings according to @zeroZshadow? )

Reply:
If you receive a component update, it has to be that either your component fields were updated or you received events (or both). In case we only receive events as part of a component update, the current idea is to emit the EventsReceived tags but not emit ComponentsUpdated tags.

Reply:
You can get a completely empty update. I'm not sure where you would do this, but anything that sends only changes could easily do this. Especially if rather than sending a timestamp, something only sends the time delta since the last update.

Reply:
To make sure I understand. Is it possible to send a completely empty component update by e.g. calling `connection.SendComponentUpdate<Foo>(new Foo.Update());`? This would bot be a trivial thing to support and I also wonder whether something like this should be conceptually permitted by the worker SDK (why not have some sort of ping event instead?) so I suggest we consciously don't support empty updates for now and start supporting them later if necessary.